app/client/ui-modules/settings-page/state/reducers/change-password-slice.ts

```diff
@@ -0,0 +1,112 @@
+import { Accounts } from "meteor/accounts-base";
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { AppDispatch, RootState } from "/app/client/store";
+
+export type ChangePasswordUIState = {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+  isLoading: boolean;
+  error: string;
+};
+
+const initialState: ChangePasswordUIState = {
+  currentPassword: "",
+  newPassword: "",
+  confirmPassword: "",
+  isLoading: false,
+  error: "",
+};
+
+export const changePassword = createAsyncThunk<
+  void,
+  void,
+  { state: RootState; dispatch: AppDispatch; rejectValue: string }
+>("settings/changePassword", async (_, { getState, rejectWithValue }) => {
+  const state = getState().changePasswordForm;
+  const { currentPassword, newPassword, confirmPassword } = state;
+
+  if (!currentPassword?.trim()) {
+    return rejectWithValue("Current password is required.");
+  }
```

Comment:
I really like that you are confirming old password. but the errors for incorrect old passwords are going silent.
my test scenario was give a very wrong current password and try to update the passwrod and when i tried to sign in again the password wasnt changed to the new one and it was still old one.

I had a look and looks like `Accounts.changePassword(currentPassword, newPassword, (err) => {` nicely catch the error but the error is not shown to the user which i think would be tied to your toast not working either